core/unix/os.c

```diff
@@ -8612,8 +8630,10 @@ post_system_call(dcontext_t *dcontext)
              * post-syscall natively anyway, there's no need to restore here.
              */
 #    endif
-            HEAP_TYPE_FREE(dcontext, dcontext->sys_param1, clone3_syscall_args_t,
-                           ACCT_OTHER, PROTECTED);
+            uint app_clone_args_size =
+                (uint)sys_param(dcontext, SYSCALL_PARAM_CLONE3_CLONE_ARGS_SIZE);
```

Comment:
Generally we avoid sys_param() in post-syscall b/c some registers (and sometimes memory) are clobbered and it's easy to make a mistake with which ones are not for different system call gateways: so we always store in the cache vars sys_paramX to avoid problems.

Reply:
Used `sys_param`.